Automattic/woocommerce-payments
# Fix monthly deposit schedule displayed incorrectly as '3rd' or '31st' when schedule set to 'Last day of the month'

Fixes #5616
Fixes #5662

#### Changes proposed in this Pull Request









This PR correctly renders the monthly deposit schedule as `monthly on the last day of the month` on the **Payments → Overview** screen when the schedule has been set to 'Last day of the month'.

<img width="687" alt="image" src="https://user-images.githubusercontent.com/3147296/221473878-048cc262-341d-44de-90d6-0fa916100760.png">

Logic for determining the monthly anchor label was used across 2 files ([1](https://github.com/Automattic/woocommerce-payments/blob/8085bf0057dc1e7c5453a7cdfd3d6be2a883daa0/client/deposits/utils/index.js/#L77-L87), [2](https://github.com/Automattic/woocommerce-payments/blob/8085bf0057dc1e7c5453a7cdfd3d6be2a883daa0/client/settings/deposits/index.js/#L39-L72)), therefore I've combined this logic into the `getDepositMonthlyAnchorLabel` function:

https://github.com/Automattic/woocommerce-payments/blob/eec5e7874dd0be58b9fb7010c1130b57007f1e4c/client/deposits/utils/index.js#L51-L54

#### Testing instructions





1. Go to **Payments → Settings**.
2. Set the deposit schedule to **Last day of the month** and click save. Also ensure all select input options are rendered correctly (1st-28th + 'Last day of the month').

   <img width="1078" alt="image" src="https://user-images.githubusercontent.com/3147296/221476668-6c3ae83a-0dfb-46da-aff6-bd43f76237ff.png">

4. Go to **Payments → Overview**.
5. Check that the deposit schedule is displayed as "Deposits set to monthly on the last day of the month."
6. Repeat steps 2-5 with various `daily`, `weekly` and `monthly` schedules, ensuring that the expected label is displayed on the **Payments → Overview** screen.

-------------------

- [x] Covered with tests **Yes**
- [x] Tested on mobile **Yes**

**Post merge**
